kube-runtime/src/utils/watch_ext.rs

```diff
@@ -36,5 +37,16 @@ pub trait WatchStreamExt: Stream {
     {
         EventFlatten::new(self, true)
     }
+
+    /// Create a [`StreamSubscribe`] from a [`watcher()`] stream
+    ///
+    /// Allows additional consumers to subscribe to the stream, without consuming the stream itself.
```

Comment:
Same doc comment as on `StreamSubscribe` should be present here as this is likely the interface users go through.

Here it also needs an example.